addons/mass_mailing/tests/test_mailing_controllers.py

```diff
@@ -393,6 +394,69 @@ def test_mailing_unsubscribe_from_list_with_update_tour(self):
+        blocklist (see tour).
+        Tour effects
+          * opt-in List3 from opt-out, opt-in List2, opt-out List1;
+          * add feedback (as new opt-out) 'My feedback';
+          * add email in block list;
+        """
+        test_feedback = "My feedback"
+        portal_user = mail_new_test_user(
+            self.env,
+            email=tools.formataddr(("Déboulonneur", "fleurus@example.com")),
+            groups='base.group_portal',
+            login='user_portal_fleurus',
+            name='Déboulonneut from Fleurus',
+            signature='--\nDéboulonneur',
+        )
+
+        for test_user in [portal_user]:  # self.user_marketing
+            with self.subTest(test_user=test_user):
+                _test_email, test_email_normalized = test_user.email, test_user.email_normalized
+                self.authenticate(test_user.login, test_user.login)
+
+                # launch 'my' mailing' tour
+                with freeze_time(self._reference_now):
+                    self.start_tour(
+                        "/mailing/my",
+                        "mailing_portal_unsubscribe_from_my",
+                        login=test_user.login,
+                    )
+
+                # fetch contact and its subscription and blacklist status, to see the tour effects
+                contact_l1 = self.mailing_list_1.contact_ids.filtered(
+                    lambda contact: contact.email == test_email_normalized
+                )
+                subscription_l1 = self.mailing_list_1.subscription_ids.filtered(
+                    lambda subscription: subscription.contact_id == contact_l1
+                )
+                contact_l2 = self.mailing_list_2.contact_ids.filtered(
+                    lambda contact: contact.email == test_email_normalized
+                )
+                subscription_l2 = self.mailing_list_3.subscription_ids.filtered(
+                    lambda subscription: subscription.contact_id == contact_l2
+                )
+                contact_l3 = self.mailing_list_3.contact_ids.filtered(
+                    lambda contact: contact.email == test_email_normalized
+                )
+                subscription_l3 = self.mailing_list_3.subscription_ids.filtered(
+                    lambda subscription: subscription.contact_id == contact_l3
+                )
+                self.assertEqual(contact_l2, contact_l3,
+                                'When creating new membership, should link with first found existing contact')
+                self.assertTrue(subscription_l1.opt_out)
+                self.assertFalse(subscription_l2.opt_out)
+                self.assertFalse(subscription_l3.opt_out)
+                for contact in (contact_l1 + contact_l3):
+                    msg_fb = contact.message_ids[0]
+                    self.assertEqual(
+                        msg_fb.body,
+                        Markup(f'<p>Feedback from {contact.email_normalized}: {test_feedback}</p>')
```

Comment:
same here, we don't need to convert to markup to compare (the email won't be escaped anyway)